pkg/kubectl/cmd/delete.go

```diff
@@ -254,7 +267,10 @@ func ReapResult(r *resource.Result, f cmdutil.Factory, out io.Writer, isDefaultD
 			// If there is no reaper for this resources and the user didn't explicitly ask for stop.
 			if kubectl.IsNoSuchReaperError(err) && isDefaultDelete {
 				// No client side reaper found. Let the server do cascading deletion.
-				return cascadingDeleteResource(info, out, shortOutput, mapper)
+				if err := deleteResourceAndWait(info, out, shortOutput, mapper, false, waitForDeletion, timeout); err != nil {
```

Comment:
Name the `false` so it's obvious that it's cascading?

Also, could you add the todo here saying we should use propagationPolicy=Foreground, so that it will wait until all the depdents are deleted? Also explain why we are not using the option today (because federation apiserver doesn't handle the finalizer)